typedspark/_utils/load_table.py

```diff
@@ -15,14 +16,26 @@
 from typedspark._utils.register_schema_to_dataset import register_schema_to_dataset
 
 
+def _replace_illegal_column_names(dataframe: DataFrame) -> DataFrame:
+    """Replaces illegal column names with a legal version."""
+    for column in dataframe.columns:
+        dataframe = dataframe.withColumnRenamed(column, _replace_illegal_characters(column))
```

Comment:
I'm not 100% sure if there is a performance impact of doing a `withColumnRenamed` on each column. But possibly cleaner to check for the condition first (existence of illegal column name) and only calling the `withColumnRenamed` if the condition is true